dascore/utils/patch.py

```
@@ -583,15 +621,17 @@ def _get_filename(path_ser, strip_extension):
-    if "path" in col_set and df["path"].astype(str).str.len().gt(0).any():
-        return _get_filename(df["path"], strip_extension)
-    # Determine the requested fields and get the ones that are there.
+    path_ser = df["path"].astype(str) if "path" in col_set else None
+    if path_ser is not None:
+        # synthetic in-memory identities are not real file names
+        usable = path_ser.str.len().gt(0) & ~path_ser.map(is_memory_uri)
+        if usable.any():
+            return _get_filename(df["path"], strip_extension)
+    # Determine the requested fields; absent columns render as empty so
+    # names don't depend on which metadata engine produced the dataframe.
     coord_fields = zip([f"{x}_min" for x in coords], [f"{x}_max" for x in coords])
```

Ruff (0.15.20)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)